projects/kit/directives/dropdown-context/dropdown-context.directive.ts

```diff
@@ -0,0 +1,163 @@
+import {
+    ComponentFactoryResolver,
+    ComponentRef,
+    Directive,
+    ElementRef,
+    forwardRef,
+    HostListener,
+    Inject,
+    Injector,
+    Input,
+    Optional,
+} from '@angular/core';
+import {
+    getClosestFocusable,
+    setNativeFocused,
+    TuiActiveZoneDirective,
+    TuiContextWithImplicit,
+    tuiDefaultProp,
+    TuiPortalService,
+} from '@taiga-ui/cdk';
+import {
+    AbstractTuiDropdown,
+    TUI_DROPDOWN_DIRECTIVE,
+    TuiDropdown,
+    TuiDropdownBoxComponent,
+} from '@taiga-ui/core';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+import {EMPTY} from 'rxjs';
+
+// @dynamic
+@Directive({
+    selector: '[tuiDropdownContext]',
+    providers: [
+        {
+            provide: TUI_DROPDOWN_DIRECTIVE,
+            useExisting: forwardRef(() => TuiDropdownContextDirective),
+        },
+    ],
+})
+export class TuiDropdownContextDirective
+    extends AbstractTuiDropdown
+    implements TuiDropdown {
+    @Input('tuiDropdownContext')
+    @tuiDefaultProp()
+    content: PolymorpheusContent = '';
+
+    readonly refresh$ = EMPTY;
+    private lastClickedClientRect: ClientRect = this.getClientRectFromDot(0, 0);
+    private hostRef: ComponentRef<TuiDropdownBoxComponent> | null = null;
+
+    constructor(
+        protected readonly elementRef: ElementRef,
+        @Inject(ComponentFactoryResolver)
+        componentFactoryResolver: ComponentFactoryResolver,
+        @Inject(Injector) injector: Injector,
+        @Inject(TuiPortalService) portalService: TuiPortalService,
+        @Optional() activeZone: TuiActiveZoneDirective | null,
+    ) {
+        super(componentFactoryResolver, injector, portalService, elementRef, activeZone);
+    }
+
+    get clientRect(): ClientRect {
+        return this.lastClickedClientRect;
+    }
+
+    get context(): TuiContextWithImplicit<() => void> {
+        return {$implicit: this.closeDropdownBox.bind(this)};
+    }
+
+    get fixed(): boolean {
+        return true;
+    }
+
+    private get dropdownContent(): HTMLElement | null {
+        return this.hostRef?.instance.contentElementRef?.nativeElement || null;
+    }
+
+    @HostListener('contextmenu.prevent', ['$event'])
+    onContextMenu(event: MouseEvent) {
+        const {clientX: x, clientY: y} = event;
+
+        this.closeDropdownBox();
+        this.hostRef = this.openDropdown(x, y);
```

Comment:
We do we need `hostRef` if we already have `dropdownBoxRef`?

Reply:
deleted